core/src/main/java/org/apache/accumulo/core/rpc/TTimeoutTransport.java

```diff
@@ -87,6 +84,12 @@ TTransport createInternal(SocketAddress addr, long timeoutMillis) throws TTransp
       socket = openSocket(addr, (int) timeoutMillis);
     } catch (IOException e) {
       // openSocket handles closing the Socket on error
+      if (e instanceof AsynchronousCloseException) {
+        if (Thread.currentThread().isInterrupted()) {
+          Thread.currentThread().interrupt();
+          throw new UncheckedIOException(e);
+        }
+      }
```

Comment:
@ctubbsii  and I talked about this issue after I put up my most recent comment. Apologies for not updating the PR. He suggested a minor tweak to make the fix more targeted to the original issue.

```suggestion
      if (e instanceof ClosedByInterruptException) {
        Thread.currentThread().interrupt();
        throw new UncheckedIOException(e);
      }
```